File: implementations/php/src/Data/ProductResearch.php
<?php

declare(strict_types=1);

namespace GraphQL\Compatibility\Data;

use GraphQL\Utils\Utils;

class ProductResearch
{
    public CaseStudy $study;

    public string $outcome;

    public function __construct(array $data)
    {
        Utils::assign($this, $data);
    }
}

File: implementations/graphql-java-kickstart/src/main/java/graphql/kickstart/federation/compatibility/model/ProductResearch.java
package graphql.kickstart.federation.compatibility.model;

import java.util.List;

public record ProductResearch(CaseStudy study, String outcome) {

    public static final ProductResearch FEDERATION_STUDY = new ProductResearch(new CaseStudy("1234", "Federation Study"), null);
    public static final ProductResearch STUDIO_STUDY = new ProductResearch(new CaseStudy("1235", "Studio Study"), null);
    public static final List<ProductResearch> RESEARCH_LIST = List.of(FEDERATION_STUDY, STUDIO_STUDY);
}
